src/PhpSpec/Wrapper/Subject/Caller.php

```diff
@@ -137,7 +137,8 @@ public function getWrappedObject(): mixed
             return $this->wrappedObject->getInstance();
         }
 
-        if (!$className || !class_exists($className)) {
+        /** @var string $className don't trust it's a class-string, we still want to check */
```

Comment:
I don't think this is needed. `class-string` need not be the name of a class. it could be an interface, in which case the class_exists check would fail.

Reply:
I'm a little unsure whether it should be class-string at all, as in some cases it's provided by the user

Reply:
That is I think it's provided by the user but we force it to class-string :/ Will address later